# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Msdff/FlyRankAiAssignment/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Refresh / Content Opportunity Scoring

Editors have hundreds of pages that may need updating, but limited time to review them all. I want to know whether signals like a page's age, content depth (word count), competition level, and how it's being discovered (organic search vs. AI referral) can help identify which pages deserve review first. This turns a vague sense of "everything looks old" into a ranked, defensible action list — which is exactly the kind of decision-support output this lane is meant to produce.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision this improves**: Which page an editor should review and refresh first, given limited time.

**Who acts on it**: A content editor or SEO team member with a fixed weekly review capacity.

**Cost of a wrong call**: If an editor spends time on a page that's already performing fine, that's wasted effort while a genuinely underperforming page — old, thin on content, or losing visibility to competition continues to lose ground unreviewed. Both errors are costly: one wastes limited editor hours, the other lets a real opportunity go unaddressed.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import os
os.chdir("..")   
print(os.getcwd())  

d:\Intenship Files\IntenshipCompanies\FlyRankAi\Assignments\Assignment No.1\GithubRepository\FlyRankAiAssignment


In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

import pandas as pd, numpy as np
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# The label: a page is 'declining' when its recent trend is down. Simple, honest starter label.
df["is_declining_label"] = df["trend_direction"].str.lower().eq("down").astype(int)
print(df.shape[0], "pages |  declining rate:", round(df["is_declining_label"].mean(), 3))
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Discovery 1: how many old pages are still getting real traffic?
old_but_active = df[(df["content_age_days"] >= 365) & (df["pageviews_90d"] >= 100)]
print("Old pages (365+ days) still getting 100+ pageviews:", len(old_but_active), "out of", len(df))

# Discovery 2: does content length relate to how many sessions a page gets?
short_pages = df[df["word_count"] < 1200]
long_pages = df[df["word_count"] >= 1200]
print("Avg sessions_90d — short pages (<1200 words):", round(short_pages["sessions_90d"].mean(), 1))
print("Avg sessions_90d — long pages (1200+ words):", round(long_pages["sessions_90d"].mean(), 1))

# Discovery 3: how many pages get any traffic from AI tools?
ai_referred = df[df["ai_sessions_90d"] > 0]
print("Pages with at least 1 AI-referred session:", len(ai_referred), "out of", len(df))

# Discovery 4: does competition level relate to click-through rate?
print(df.groupby("competition_level")["ctr"].mean().round(2))



30000 pages |  declining rate: 0.542
Old pages (365+ days) still getting 100+ pageviews: 365 out of 30000
Avg sessions_90d — short pages (<1200 words): 3.3
Avg sessions_90d — long pages (1200+ words): 42.2
Pages with at least 1 AI-referred session: 1930 out of 30000
competition_level
HIGH      0.26
LOW       0.34
MEDIUM    0.24
Name: ctr, dtype: float64


**What this shows:**

Out of 30,000 pages in the starter dataset, 54.2% currently show a declining trend confirming that decline is a common, not rare, pattern worth prioritizing.

365 pages are both old (365+ days) and still generating meaningful traffic (100+ pageviews in the last 90 days) these are strong refresh candidates, since real audience attention already exists on them even as they age.

Content length shows a striking relationship with engagement: short pages (under 1,200 words) average just 3.3 sessions over 90 days, while longer pages (1,200+ words) average 42.2 sessions — over 12x higher. This suggests word count could be a meaningful signal for prioritization, though it may also reflect deeper factors like topic type or search intent that are worth investigating further.

Only 1,930 pages (about 6.4%) show any AI-referred traffic at all, confirming the lane guide's warning that this signal is sparse — useful for exploratory patterns, but not reliable enough to build a standalone model on.

Click-through rate also varies by competition level: low-competition pages have the highest average CTR (0.34), while medium-competition pages have the lowest (0.24) — suggesting competition context matters when judging whether a page's CTR is actually underperforming or normal for its category.

Together, these numbers confirm multiple independent signals exist in the starter dataset decline rate, content depth, traffic longevity, and competition context that a priority score could meaningfully combine over the next seven weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What this work can say: This project can produce observed and directional findings — for example, which combinations of age, content depth, and competition context are associated with lower engagement or visibility, and which pages represent the strongest decision-support candidates for review. Results describe patterns in the data, not guarantees about outcomes.

What this work cannot say: This project cannot claim to predict Google's ranking algorithm, cannot prove that refreshing a page will cause it to recover (that would require a controlled experiment), and cannot treat any single score as certainty. All recommendations are meant to guide human review, not replace it.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.